DependencyInjection/SonataAdminExtension.php

```diff
@@ -214,13 +214,47 @@ public function prepend(ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
 
-        if (isset($bundles['JMSDiExtraBundle'])) {
+        if (!isset($bundles['JMSDiExtraBundle'])) {
+            return;
+        }
+
+        $configs = $container->getExtensionConfig($this->getAlias());
+        $config = $this->processConfiguration(new Configuration(), $configs);
+        if (!$config['options']['enable_jms_di_extra_autoregistration']) {
```

Comment:
I've added it a little bit further down the code so that it doesn't get triggered when the user has already added the necessary pattern manually in the jms_di_extra configuration.